docs/OpenShift GitOps CLI User Guide.md

```diff
@@ -0,0 +1,373 @@
+```
+**Note:** The above output is just for reference. The actual details might be different based on the version of OpenShift GitOps argocd CLI client installed.
+
+### Installing the Red Hat OpenShift GitOps CLI on Windows
+#### Prerequisites
+- ZIP file tools like winrar, 7zip etc
+#### Procedure
+- Download the [CLI tool](https://mirror.openshift.com/pub/openshift-v4/clients/argocd-cli/1.12.0/argocd-cli-windows-amd64.zip).
+- Extract the archive with a ZIP program.
+- Add the location of your `argocd` files to your `PATH` environment variable.
+#### Verification
+Run the following command to validate that the installation has succeeded.
+```
+argocd version --client
+```
+Sample output:
+```
+argocd: v2.9.2+c5ea5c4
+  BuildDate: 2023-12-18T12:35:23Z
+  GitCommit: c5ea5c4df52943a6fff6c0be181fde5358970304
+  GitTreeState: clean
+  GoVersion: go1.20.10
+  Compiler: gc
+  Platform: linux/amd64
+  ExtraBuildInfo: openshift-gitops-version: 1.11.0, release: 0718122023
+```
+**Note:** The above output is just for reference. The actual details might be different based on the version of OpenShift GitOps argocd CLI client installed.
+
+### Installing the Red Hat OpenShift GitOps CLI on macOS
+#### Prerequisites
+- tar
+#### Procedure
+- Download the CLI tool for the appropriate processor architecture
+    - [macOS on Intel](https://mirror.openshift.com/pub/openshift-v4/clients/argocd-cli/1.12.0/argocd-cli-darwin-amd64.tar.gz)
+    - [macOS on ARM](https://mirror.openshift.com/pub/openshift-v4/clients/argocd-cli/1.12.0/argocd-cli-darwin-arm64.tar.gz)
+
+- Extract the archive with a ZIP program.
+- Add the location of your `argocd` files to your `PATH` environment variable.
+#### Verification
+Run the following command to validate that the installation has succeeded.
+```
+argocd version --client
+```
+Sample output:
+```
+argocd: v2.9.2+c5ea5c4
+  BuildDate: 2023-12-18T12:35:23Z
+  GitCommit: c5ea5c4df52943a6fff6c0be181fde5358970304
+  GitTreeState: clean
+  GoVersion: go1.20.10
+  Compiler: gc
+  Platform: linux/amd64
+  ExtraBuildInfo: openshift-gitops-version: 1.11.0, release: 0718122023
+```
+**Note:** The above output is just for reference. The actual details might be different based on the version of OpenShift GitOps argocd CLI client installed.
+
+## OpenShift GitOps argocd reference
+This section lists the basic `argocd` CLI commands.
+**Note** MicroShift based installation do not host an ArgoCD server and supports only the `core` mode of execution. 
+In the `core` mode (`--core` argument specified), the CLI talks directly to the Kubernetes API server set as per the `KUBECONFIG` environment variable or the default file `$HOME/.kube/config`. There is no need for users to login into the ArgoCD server for executing commands.
+
+### Basic syntax
+- Normal mode
+  ```
+  argocd [command or options] [arguments…​]
+  ```
+
+- Core mode
+  ```
+  KUBECONFIG=~/.kube/config argocd --core [command or options] [arguments…​]
```

Comment:
@anandf Suggest including a description and purpose of both the modes.

Reply:
Added a short text describing what the two modes mean. and when to use what mode.